--- config/config.php	2010-08-27 10:55:53.000000000 +0200
+++ config/config.php.new	2010-08-27 10:58:18.000000000 +0200
@@ -118,11 +118,12 @@ public function loadConfig() {
 	$config['gearsURL'] = 'http://tools.google.com/gears/';
 
 	// database settings	
-	$config['pg_host'] = 'localhost';
-	$config['pg_database'] = 'filesender';
-	$config['pg_port'] = '5432';
-	$config['pg_username'] = 'postgres';
-	$config['pg_password'] = 'yoursecretpassword';
+	include('/etc/filesender/debian-db.php');
+	$config['pg_host'] = empty($dbserver) ? 'localhost' : $dbserver;
+	$config['pg_database'] = empty($dbname) ? 'filesender' : $dbname;
+	$config['pg_port'] = empty($dbport) ? '5432' : $dbport;
+	$config['pg_username'] = empty($dbuser) ? 'filesender' : $dbuser;
+	$config['pg_password'] = empty($dbpass) ? 'filesender' : $dbpass;
 
 	// email
 	$config['default_emailsubject'] = "{siteName}: {filename}";
