--- config/config.php	2012-07-15 11:00:36.988305035 +0200
+++ config/config.php	2012-07-15 22:27:39.304063050 +0200
@@ -123,13 +123,14 @@
 	$config['site_simplesamllocation'] = '/usr/share/simplesamlphp/';
 	$config['log_location'] = '/usr/share/filesender/log/';
 
-	$config["db_type"] = "pgsql";// pgsql or mysql
-	$config['db_host'] = 'localhost';
-	$config['db_database'] = 'filesender';
-	$config['db_port'] = '5432';
+	include('/etc/filesender/debian-db.php');
+	$config["db_type"] = empty($dbtype) ? 'pgsql' : $dbtype;
+	$config['db_host'] = empty($dbserver) ? 'localhost' : $dbserver;
+	$config['db_database'] = empty($dbname) ? 'filesender' : $dbname;
+	$config['db_port'] = empty($dbport) ? '5432' : $dbport;
 	// database username and password
-	$config['db_username'] = 'filesender';
-	$config['db_password'] = 'yoursecretpassword';
+	$config['db_username'] = empty($dbuser) ? 'filesender' : $dbuser;
+	$config['db_password'] = empty($dbpass) ? 'filesender' : $dbpass;
 
 	//Optional DSN format overides db_ settings
 	//$config['dsn'] = "pgsql:host=localhost;dbname=filesender";
