--- config/config.php	2011-12-15 17:48:01.023376262 +0100
+++ config/config.php.new	2011-12-17 11:21:34.337624961 +0100
@@ -127,12 +127,13 @@
 	$config['log_location'] = '/usr/share/filesender/log/';	
 
 	$config["db_type"] = "pgsql";// pgsql or mysql
-	$config['db_host'] = 'localhost';
-	$config['db_database'] = 'filesender';
-	$config['db_port'] = '5432';
+	include('/etc/filesender/debian-db.php');
+	$config['db_host'] = empty($dbserver) ? 'localhost' : $dbserver;
+	$config['db_database'] = empty($dbname) ? 'filesender' : $dbname;
+	$config['db_port'] = empty($dbport) ? '5432' : $dbport;
 	// database username and password
-	$config['db_username'] = 'filesender';
-	$config['db_password'] = 'yoursecretpassword';
+	$config['db_username'] = empty($dbuser) ? 'filesender' : $dbuser;
+	$config['db_password'] = empty($dbpass) ? 'filesender' : $dbpass;
 	
 	//Optional DSN format overides db_ settings
 	//$config['dsn'] = "pgsql:host=localhost;dbname=filesender";
